src/Symfony/Component/Security/Http/LoginLink/LoginLinkHandler.php

```diff
@@ -71,7 +75,15 @@ public function createLoginLink(UserInterface $user): LoginLinkDetails
 
     public function consumeLoginLink(Request $request): UserInterface
     {
-        $username = $request->get('user');
+        $usernameMessage = $request->get('user');
+        try {
+            $username = $this->encryption->decrypt($usernameMessage);
+        } catch (MalformedCipherException $exception) {
+            // Keep compatibility with older versions of the code
+            $username = $usernameMessage;
```

Comment:
@Nyholm what about explicitly indicate a deprecation?

because from "now" on, given https://github.com/symfony/symfony/pull/39275/files#diff-9a769cb79d1b50d804969e256e0443b7d98bc0e6bf421778ad3b22aa02849507R62, the `user` parameter will be always encoded and should never enter here

also, I was wondering if the solution of a `MalformedCipherException ` should or not be `$username = $usernameMessage`. In other words, a cipher could be malformed for many reasons, not only that `user` parameter is passed in clear

Reply:
could we think about a `UsernameNotEncryptedException` ?

Reply:
I thought about this too. 
But I guess that people would generate a link, then upgrade and then user consumes the link. I just want to make sure the link works. If we didn't care about BC, I would always throw `InvalidLoginLinkException` here. Doing `$username = $usernameMessage` would also lead to a `InvalidLoginLinkException` if the user parameter wasnt encrypted. 

Sure, I can trigger a deprecation notice.